nimbus-ui/nimbusui/src/app/components/platform/form/elements/picklist.component.ts

```diff
@@ -169,10 +198,34 @@ export class OrderablePickList extends BaseElement implements OnInit, ControlVal
         } else {
             this.selectedOptions = [];
             this.targetList.forEach(element => {
-                this.selectedOptions.push(element.code);
+                if (element.code) {
+                    this.selectedOptions.push(element.code);
+                } else {
+                    this.selectedOptions.push(element);
+                }
             });
             this.value = this.selectedOptions;
         }
+        /* comes into this loop when leafState is not null and new values are added
+         prime-ng adds data of type Values{code, label} to leafState since the [source] is of type {element.values} 
+         ex: leafState : ["Mon","Tue",{'code' : "Wed", 'label' : "Wednesday"}]
+         The below logic, transforms the data into just codes i.e leafState : ["Mon","Tue","Wed"] 
+        */
+        if (this.element.leafState) {
+            this.element.leafState.forEach((state, i) => {
+                if (state && state.code) {
+                    const code = state.code;
+                    this.element.leafState.splice(i, 1, code);
+                }
+            });
+        } else {
+            if (this.value) {
+                this.element.leafState = this.value;
+            }
+        }
```

Comment:
Consider moving this code block and documentation into a private method.

Reply:
Addressed